src/hooks/api/usePrices.ts

```diff
@@ -10,10 +10,20 @@ import fetcher from './fetcher';
 /**
  * Returns prices for a given commodity
  */
-export function usePrices(c: Commodity | undefined): UseQueryResult<PriceDBMap> {
+export function usePrices(params: {
+  from?: Commodity,
+  to?: Commodity,
+}): UseQueryResult<PriceDBMap> {
   return useQuery({
-    queryKey: [Price.CACHE_KEY, { commodity: c?.guid }],
-    queryFn: fetcher(() => getPrices({ from: c }), `${Price.CACHE_KEY}/${c?.guid}`),
-    enabled: !!c,
+    queryKey: [Price.CACHE_KEY, { from: params.from?.guid, to: params.to?.guid }],
+    queryFn: fetcher(
+      () => getPrices(params),
+      `${Price.CACHE_KEY}/${params.from?.guid}.${params.to?.guid}`,
+    ),
+    enabled: (
+      !!('from' in params && params.from)
+      || !!('to' in params && params.to)
+      || (!('from' in params) && !('to' in params))
```

Comment:
this is to make sure we don't trigger the query when we pass from or to explicitly but with `undefined` value. This usually means the commodity hasn't been loaded yet